### PR TITLE
Windows support

### PR DESCRIPTION
**Pull request by wjwwood**
This pull request fixes Windows related issues which prevent all of ROS 2 from compiling.

This is a work in progress.

Connects to ros2/ros2#9

### CODE REVIEW

**Comment by tfoote** [comment]
lgtm

**Comment by esteve** [comment]
+1

File: connext_cmake_module/CMakeLists.txt [hunk 1]
```diff
@@ -1,9 +1,21 @@
 cmake_minimum_required(VERSION 2.8.3)
 
-project(connext_cmake_module NONE)
+project(connext_cmake_module)
```
Comment (dirk-thomas) [inline connext_cmake_module/CMakeLists.txt]: Why remove `NONE` here? The package does seem to compile anything.

File: rmw_connext_xtypes_dynamic_cpp/CMakeLists.txt [hunk 2]
```diff
@@ -24,7 +26,7 @@ ament_export_dependencies(rmw rosidl_generator_cpp)
 
 register_rmw_implementation()
 
-link_directories(${Connext_LIBRARY_DIRS})
+link_directories("${Connext_LIBRARY_DIRS}")
```
Comment (dirk-thomas) [inline rmw_connext_xtypes_dynamic_cpp/CMakeLists.txt]: Why does this need quotes?

Comment (wjwwood) [inline rmw_connext_xtypes_dynamic_cpp/CMakeLists.txt]: If one of the paths contains whitespace they will be interpreted as two paths. E.g. a value of `C:\foo;C:\Program Files\bar;C:\baz` would be considered `link_directories("C:\foo;C:\Program" "Files\bar;C:\baz")` which would produce this when iterated on: `C:\foo`, `C:\Program`, `Files\bar`, and `C:\baz`.

At least that's what I observed, maybe there is a better way to fix it.

File: connext_cmake_module/CMakeLists.txt (hunk 1 above)
Comment (wjwwood) [inline connext_cmake_module/CMakeLists.txt]: If you put `NONE` there then this variable `CMAKE_SIZEOF_VOID_P` is not set and you cannot include the `FindConnext.cmake` file in order to fill out the env hook.

Comment (dirk-thomas) [inline connext_cmake_module/CMakeLists.txt]: It looks like `FindConnext.cmake` should make sure to perform the necessary compile checks in order to be used from this with `NONE` being set.

File: rmw_connext_xtypes_dynamic_cpp/CMakeLists.txt (hunk 2 above)
Comment (dirk-thomas) [inline rmw_connext_xtypes_dynamic_cpp/CMakeLists.txt]: Sounds reasonable. I guess that might apply to other locations too. Will have to try this once we have the CI job running, e.g. always build in a location with spaces (@tfoote fyi).

File: connext_cmake_module/CMakeLists.txt (hunk 1 above)
Comment (dirk-thomas) [inline connext_cmake_module/CMakeLists.txt]: I thought `enable_language` would be the way to go but the docs suggest that it doesn't set the desired variables. Just adding a comment here why we can't use `NONE` should be good for now.

**Pull request merged**